controller/tap/server.go

```diff
@@ -94,13 +99,18 @@ func (s *GRPCTapServer) TapByResource(req *pb.TapByResourceRequest, stream pb.Ta
 	}
 
 	if len(pods) == 0 {
-		resType := res.GetType()
-		resName := res.GetName()
-		if foundDisabledPods {
-			return status.Errorf(codes.NotFound,
-				"all pods found for %s/%s have tapping disabled", resType, resName)
+		errStrings := []string{}
+		errStr := fmt.Errorf("no pods to tap for %s/%s", res.GetType(), res.GetName())
+		errStrings = append(errStrings, errStr.Error())
+		if tapDisabled {
+			errStr = fmt.Errorf("pods found with tap disabled via the %s annotation", pkgK8s.ProxyDisableTapAnnotation)
+			errStrings = append(errStrings, errStr.Error())
 		}
-		return status.Errorf(codes.NotFound, "no pods found for %s/%s", resType, resName)
+		if tapNotEnabled {
+			errStr = fmt.Errorf("pods found with tap not enabled; try restarting resource so that it can be injected")
```

Comment:
Once this is updated, I can update the error message to instead ask for `viz check` where the restart injector message can be displayed full list of failed injected pods.